apps/public-docsite-v9/src/Concepts/Slots/Slots.stories.mdx

```diff
@@ -0,0 +1,379 @@
+
+```tsx
+<Button icon={<CalendarRegular24 />}
+```
+
+### Passing props
+
+When a slot renders a component or element by default, you can pass props to it.
+
+The `Avatar` component has a `badge` slot that renders a `PresenceBadge`.
+This allows you pass `PresenceBadge` props.
+
+```tsx
+<Avatar name="Support" badge={{ status: 'available', 'aria-label': 'available' }} />
+```
+
+You can pass the `className` prop to customize the style of a slot.
+The `Avatar` badge slot supports a `className`, allowing the badge style to be customized.
+
+```tsx
+const useStyles = makeStyles({
+  badge: { color: tokens.colorBrandStroke1 },
+});
+
+const BusyBrandAvatar = () => {
+  const styles = useStyles();
+  return <Avatar name="IT probably" badge={{ status: 'busy', className: styles.badge }} />;
+};
+```
+
+You can pass the `as` prop to change the element type of a slot.
+Note that you must choose from one of the available element types the slot supports.
+
+```tsx
+<AccordionHeader as="h1">Accordion Header as h1</AccordionHeader>
+```
+
+### Replacing the entire slot
+
+When you pass content or props to a slot,
+the component renders the content within a component or element based on the slot type.
+If you need to replace the slot's entire content, including the containing element,
+pass a render function as the children.
+
+This is an escape hatch in the slots API, so prefer the other techniques whenever possible.
+
+By passing `renderBigLetterIcon` as the `children`, the `span` that normally contains the icon is replaced with an `h1`.
+
+```tsx
+const renderBigLetterIcon () => {
+  return <h1>B<h1>;
+};
+
+<Button icon={{ children: renderBigLetterIcon }}>utton</Button>;
+```
+
+## For component developers
+
+### Slot type
+
+In `@fluentui/react-utilities`, `compose/types.ts` defines the types for the slots API.
+
+> This explanation of the types starts with some oversimplification then adds more detail.
+> For full details, see the `compose/types.ts` file comments.
+
+The `Slot` type allows components to define a slot.
+
+```ts
+type Slot<Type, AlternateAs>
+```
+
+`Type` defines the default element or component for the slot.
+It can be a single intrinsic element type (i.e. an HTML element like 'div') or a React component type.
+
+`AlternateAs` allows the slot to support other kinds of elements through the `as` prop.
+It can be a single intrinsic element type, or a union of element types.
+
+Currently, `AlternateAs` only supports intrinsic element types.
+This is necessary to ensure components restrict slots where `Type` is a component type.
+Substituting other component types has caused deep typing, accessibility, and event handler problems in the past.
+
+For example,
+
+- `Slot<'div'>` means by default the slot will render a `div` element.
+- `Slot<typeof Button>` means by default the slot will render a `Button` component.
+- `Slot<'span', 'div' | 'pre'>` means that caller can use `as`to change from `span` to `div` or `pre`.
+
+### Slot definition
```

Comment:
IMO this section is currently confusing. I don't know what do you think about it, but I would completely skip it for now.

Reply:
Feedback from partners reading the slots documentation is that they need a better explanation of the slot types and what they mean when inspecting components.  We should get something documented and then clarify if partners find it confusing rather than not giving them any information.